optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -421,6 +422,10 @@ private OptaPlannerGizmoInfo generateDomainAccessors(SolverConfig solverConfig,
                 case FIELD: {
                     FieldInfo fieldInfo = annotatedMember.target().asField();
                     ClassInfo classInfo = fieldInfo.declaringClass();
+                    // SolutionDescriptor PLANNING_SCORE is also picked up as a candidate, which cause problems
+                    if (classInfo.name().toString().startsWith(SolutionDescriptor.class.getName())) {
```

Comment:
Did it happen just by changing to `BuildProducer<GeneratedBeanBuildItem>` in the `generateDomainAccessors()`'s parameters?

Reply:
Also to avoid duplication, a simple `isSolutionDescriptor()` would help.

Reply:
Nope; it to make it injectable (so an `Instance<...>` is able to create it and we don't need to use reflection to create it).